### PR TITLE
Support `lastModifiedBy` for GCS + S3

### PR DESCRIPTION
**Pull request by ajordens**

### CODE REVIEW

File: front50-gcs/src/main/java/com/netflix/spinnaker/front50/model/GcsStorageService.java [hunk 1]
```diff
@@ -237,6 +238,8 @@ public void deleteObject(String objectKey, String daoTypeName) {
 
   public <T extends Timestamped>
          void storeObject(String objectKey, String daoTypeName, T obj) {
+    obj.setLastModifiedBy(AuthenticatedRequest.getSpinnakerUser().orElse("anonymous"));
```
Comment (ajordens) [inline front50-gcs/src/main/java/com/netflix/spinnaker/front50/model/GcsStorageService.java]: Didn't test this but it should work (analogous to S3)

File: front50-gcs/src/main/java/com/netflix/spinnaker/front50/model/ServiceAccountBucketDAO.java [hunk 2]
```diff
@@ -41,7 +41,6 @@ public void update(String id, ServiceAccount permission) {
 
   private ServiceAccount upsert(String id, ServiceAccount permission) {
     permission.setName(id);
-    permission.setLastModified(System.currentTimeMillis());
```
Comment (ajordens) [inline front50-gcs/src/main/java/com/netflix/spinnaker/front50/model/ServiceAccountBucketDAO.java]: These are unnecessary calls as `lastModified` is determined by the object timestamp in GCS (and S3).

File: front50-web/src/main/groovy/com/netflix/spinnaker/front50/config/Front50WebConfig.groovy [hunk 3]
```diff
@@ -45,6 +48,11 @@ public class Front50WebConfig extends WebMvcConfigurerAdapter {
   }
 
   @Bean
+  Filter authenticatedRequestFilter() {
+    new AuthenticatedRequestFilter(true)
```
Comment (ajordens) [inline front50-web/src/main/groovy/com/netflix/spinnaker/front50/config/Front50WebConfig.groovy]: This is the filter that actually extracts the auth headers and sticks them in the MDC context.

**Comment by duftler** [comment]
@ewiseblatt

**Comment by ajordens** [comment]
FYI @anotherchrisberry 

@ewiseblatt I updated GCS (_it was an identical change to S3_) but didn't test it out.
@gregturn I _did not_ update the Redis implementation ... it felt boilerplate to do it in every single DAO. Was thinking it might be more elegant to have a serializer capable of setting `lastModifiedBy` (and `lastModified`) but figured you might have an even better spot.

**Comment by cfieber** [comment]
LGTM

**Pull request merged**